tests/route/test_default_route.py

```diff
@@ -0,0 +1,81 @@
+import ipaddress
+import re
+import pytest
+import logging
+
+logger = logging.getLogger(__name__)
+
+def test_default_route_set_src(duthost):
+    """check if ipv4 and ipv6 default src address match Loopback0 address
+
+admin@vlab-01:~$ ip route list match 0.0.0.0
+default proto bgp src 10.1.0.32 metric 20
+        nexthop via 10.0.0.57 dev PortChannel0001 weight 1
+        nexthop via 10.0.0.59 dev PortChannel0002 weight 1
+        nexthop via 10.0.0.61 dev PortChannel0003 weight 1
+        nexthop via 10.0.0.63 dev PortChannel0004 weight 1
+
+admin@vlab-01:~$ ip -6 route list match ::
+default proto bgp src fc00:1::32 metric 20
+        nexthop via fc00::72 dev PortChannel0001 weight 1
+        nexthop via fc00::76 dev PortChannel0002 weight 1
+        nexthop via fc00::7a dev PortChannel0003 weight 1
+        nexthop via fc00::7e dev PortChannel0004 weight 1 pref medium
+    """
+
+    config_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
+
+    lo_ipv4 = None
+    lo_ipv6 = None
+    los = config_facts.get("LOOPBACK_INTERFACE", {})
+    logger.info("Loopback IPs: {}".format(los))
+    for k, v in los.items():
+        if k == "Loopback0":
+            for ipstr in v.keys():
+                ip = ipaddress.ip_interface(ipstr)
+                if ip.version == 4:
+                    lo_ipv4 = ip
```

Comment:
Should lo_ipv4 and lo_ipv6 be a list to accommodate multiple loopback addresses?

Reply:
no. not heeded.  two routes only requires two src ip